network/p2p/inspector/validation/control_message_validation_inspector.go

```diff
@@ -745,27 +814,20 @@ func (c *ControlMsgValidationInspector) checkClusterPrefixHardThreshold(nodeID f
 }
 
 // logAndDistributeErr logs the provided error and attempts to disseminate an invalid control message validation notification for the error.
-func (c *ControlMsgValidationInspector) logAndDistributeAsyncInspectErrs(req *InspectRPCRequest, ctlMsgType p2pmsg.ControlMessageType, err error) {
+func (c *ControlMsgValidationInspector) logAndDistributeAsyncInspectErrs(req *InspectRPCRequest, ctlMsgType p2pmsg.ControlMessageType, errs p2p.InvCtrlMsgErrs) {
 	lg := c.logger.With().
 		Bool(logging.KeySuspicious, true).
 		Bool(logging.KeyNetworkingSecurity, true).
 		Str("peer_id", p2plogging.PeerId(req.Peer)).
 		Logger()
 
-	switch {
-	case IsErrActiveClusterIDsNotSet(err):
-		lg.Warn().Err(err).Msg("active cluster ids not set")
-	case IsErrUnstakedPeer(err):
-		lg.Warn().Err(err).Msg("control message received from unstaked peer")
-	default:
-		err = c.distributor.Distribute(p2p.NewInvalidControlMessageNotification(req.Peer, ctlMsgType, err))
-		if err != nil {
-			lg.Error().
-				Err(err).
-				Msg("failed to distribute invalid control message notification")
-		}
-		lg.Error().Err(err).Msg("rpc control message async inspection failed")
+	err := c.distributor.Distribute(p2p.NewInvalidControlMessageNotification(req.Peer, ctlMsgType, errs))
+	if err != nil {
+		lg.Error().
```

Comment:
```suggestion
		lg.Fatal.
```
From `Distribute` [godoc](https://github.com/onflow/flow-go/blob/64c76434fb3aa275175816a76d05c7d2f575f500/network/p2p/consumers.go#L91-L94): `Any error returned by the distributor is non-recoverable and will cause the node to crash.`

Reply:
https://github.com/onflow/flow-go/pull/4978/commits/a14d41e67631771f4e826811f249c68d5c3739ac